common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -52,20 +50,24 @@
 import static com.microsoft.identity.common.exception.ErrorStrings.CREDENTIAL_IS_SCHEMA_NONCOMPLIANT;
 
 public class MsalOAuth2TokenCache
-        extends OAuth2TokenCache<MicrosoftStsOAuth2Strategy, MicrosoftStsAuthorizationRequest, MicrosoftStsTokenResponse>
-        implements IShareSingleSignOnState<MicrosoftAccount, MicrosoftRefreshToken> {
+        <GenericOAuth2Strategy extends OAuth2Strategy,
+                GenericAuthorizationRequest extends AuthorizationRequest,
+                GenericTokenResponse extends TokenResponse,
+                GenericAccount extends com.microsoft.identity.common.Account,
+                GenericRefreshToken extends com.microsoft.identity.common.internal.providers.oauth2.RefreshToken>
+        extends OAuth2TokenCache<GenericOAuth2Strategy, GenericAuthorizationRequest, GenericTokenResponse>
+        implements IShareSingleSignOnState<GenericAccount, GenericRefreshToken> {
 
     private static final String TAG = MsalOAuth2TokenCache.class.getSimpleName();
 
-    private List<IShareSingleSignOnState> mSharedSsoCaches; //NOPMD Suppressing PMD warning for unused variable
     private IAccountCredentialCache mAccountCredentialCache;
 
-    private IAccountCredentialAdapter<
-            MicrosoftStsOAuth2Strategy,
-            MicrosoftStsAuthorizationRequest,
-            MicrosoftStsTokenResponse,
-            MicrosoftAccount,
-            MicrosoftRefreshToken> mAccountCredentialAdapter;
+    private final IAccountCredentialAdapter<
```

Comment:
The adapter type now accepts [bounded params](https://docs.oracle.com/javase/tutorial/java/generics/bounded.html)